server/config.ts

```diff
@@ -133,6 +133,7 @@ export default {
     showAnalytics: flag('FEATURE_SHOW_ANALYTICS', false),
     showPcAnalytics: flag('FEATURE_SHOW_PC_ANALYTICS', false),
   },
+  analyticsCacheExpiryMinutes: Number(get('ANALYTICS_CACHE_EXPIRY_MINUTES', 120)),
```

Comment:
Thinking there are a couple of scenarios where having a cache expiry impacts the user experience because they would need to wait 2 hours for the data to reflect the latest extract
- Daily extract is typically produced around 2pm and not visible until 4pm
- If there was an issue with the extract and the analytics team needed to the re-run the job then again there would be a delay in seeing the latest data
  - This is a failure scenario rather than BAU so perhaps in this case we would manually restart the pods to clear the caches, but that would add operational overhead

It's probably worth mentioning this to the rest of the squad so they're aware of the impact?

One option is to
1. Include `modified` in `cacheKey`
2. Check S3 for every request and find compare the latest `modified` vale and compare it with the cache

Reply:
Bear in mind that access to data is kinda random so the new data not being visible until 4pm is a **worst** case scenario rather than always happening (if previous value happened to be cached just 1 minute before new data, e.g. 13:59pm).

It's a cache so it's to avoid an expensive operation but it means data may be a bit out of date. Hence I wasn't too aggressive with the time this was cached. We could reduce this to 1h and maybe that's good enough.

I don't think option 1 is possible because `modified` is not known by the client and we cannot include it in the `cacheKey`.

Second option means we have to make an HTTP request to S3 every time to check if the cache is fresh (bear in mind in some pages we show 2-4 graphs). We _could_ do it, but bear in mind that currently `findTable()` also gets the data so to change the logic this will have to be split, not sure how easy that is 🤔 (at the very least there are quite a few tests to update too)

The simplest thing would be to maybe reduce the cache validity to 1h and accept that "old" data could be cached/shown for up to ~1h (worst case scenario, I'd imagine on average be ~half of that).

The alternative is more involved:
1. split `AnalyticsService.findTable()` into 2 methods
  1a. one that returns that returns the latest key/metadata
  2a. one that gets the actual content
2. update cache logic to always check latest modified and if current value has a lower modified update the cache (effectively remove the existing logic/setting to generate/check expiry time of cached value)

Reply:
That's a good point - my comment was focusing on the worst case scenario 😄 

If we aren't too concerned with potentially showing out of date data then keeping the cache logic simple may be a better option. Probably helpful getting some input from rest of the squad

Reply:
I'll mention this to others in the team. I wanna experiment a bit with refactoring the `findTable()` method and the caching logic (to compare new modified vs cached modified).